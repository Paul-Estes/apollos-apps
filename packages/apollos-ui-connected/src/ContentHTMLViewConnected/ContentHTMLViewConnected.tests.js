import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';
import {
  IntrospectionFragmentMatcher,
  InMemoryCache,
} from 'apollo-cache-inmemory';
import introspectionQueryResultData from '../utils/fragmentTypes.json';

import { Providers } from '../utils/testUtils';
import GET_CONTENT_ITEM_CONTENT from './getContentItemContent';

import ContentHTMLViewConnected from './ContentHTMLViewConnected';

const fragmentMatcher = new IntrospectionFragmentMatcher({
  introspectionQueryResultData,
});

describe('ContentHTMLViewConnected component', () => {
  it('renders ContentHTMLViewConnected', async () => {
    const mock = {
      request: {
        query: GET_CONTENT_ITEM_CONTENT,
        variables: { contentId: '1' },
      },
      result: {
        data: {
          node: {
            __typename: 'DevotionalContentItem',
            id: '1',
            htmlContent: '<b>Some content!</b>',
          },
        },
      },
    };
    const tree = renderer.create(
      <Providers
        mocks={[mock]}
        cache={
          new InMemoryCache({
            fragmentMatcher,
          })
        }
      >
        <ContentHTMLViewConnected contentId={'1'} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
