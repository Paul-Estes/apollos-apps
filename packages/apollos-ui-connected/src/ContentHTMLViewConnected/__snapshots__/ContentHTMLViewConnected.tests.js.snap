// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHTMLViewConnected component renders ContentHTMLViewConnected 1`] = `
<View
  style={
    Object {
      "paddingBottom": 18,
    }
  }
>
  <View>
    <Text
      bold={false}
      italic={false}
      style={
        Object {
          "color": "#303030",
          "fontFamily": "InterUI-Regular",
          "fontSize": 16,
          "lineHeight": 26,
        }
      }
    >
      <Text
        bold={true}
        italic={false}
        style={
          Object {
            "color": "#303030",
            "fontFamily": "InterUI-Bold",
            "fontSize": 16,
            "lineHeight": 26,
          }
        }
      >
        <Text>
          Some content!
        </Text>
      </Text>
    </Text>
  </View>
</View>
`;
